dask_cuda/proxify_device_objects.py

```diff
@@ -22,41 +23,50 @@ def proxify_device_objects(
     ----------
     obj: Any
         Object to search through or wrap in a ProxyObject.
-    proxied_id_to_proxy: Dict[int, ProxyObject]
+    proxied_id_to_proxy: MutableMapping[int, ProxyObject]
         Dict mapping the id() of proxied objects (CUDA device objects) to
         their proxy and is updated with all new proxied objects found in `obj`.
     found_proxies: List[ProxyObject]
         List of found proxies in `obj`. Notice, this includes all proxies found,
         including those already in `proxied_id_to_proxy`.
+    excl_proxies: bool
+        Don't add found objects that are already ProxyObject to found_proxies.
 
     Returns
     -------
     ret: Any
         A copy of `obj` where all CUDA device objects are wrapped in ProxyObject
     """
-    return dispatch(obj, proxied_id_to_proxy, found_proxies)
+    return dispatch(obj, proxied_id_to_proxy, found_proxies, excl_proxies)
 
 
 def proxify(obj, proxied_id_to_proxy, found_proxies, subclass=None):
     _id = id(obj)
     if _id in proxied_id_to_proxy:
         ret = proxied_id_to_proxy[_id]
+        finalize = ret._obj_pxy.get("external_finalize", None)
+        if finalize:
+            finalize()
```

Comment:
Any chance `finalize` can be non-callable? Should the condition be `if callable(finalize):` perhaps and emit an error when is `not callable(finalize) and finalize is not None`?

Reply:
The `finalize` [is always callable](https://docs.python.org/3/library/weakref.html#weakref.finalize) if it isn't `None`.

Reply:
It should always be callable, I'm asking if there's any chance it `external_finalize` could eventually be something other than a callable and whether it's worth checking that. This is probably me just being overly concerned, so I thought I would check with you.

Reply:
I think that is too paranoid :) 
`_obj_pxy["external_finalize"]` is only assigned one place: https://github.com/rapidsai/dask-cuda/blob/d8f371572dd38997d909acef2d51a406a23f4824/dask_cuda/proxify_host_file.py#L219

Reply:
Fair enough.